ops/testing.py

```diff
@@ -1804,8 +1865,28 @@ def action_log(self, message):  # type:ignore
     def action_fail(self, message=''):  # type:ignore
         raise NotImplementedError(self.action_fail)  # type:ignore
 
-    def network_get(self, endpoint_name, relation_id=None):  # type:ignore
-        raise NotImplementedError(self.network_get)  # type:ignore
+    def network_get(self, binding_name: str, relation_id: Optional[int] = None) -> '_NetworkDict':
+        if relation_id is None:
+            data = self._networks.get((binding_name, None))
+            if data is not None:
+                # Get the default data if it exists
+                return data
+            # Otherwise get the data with the lowest relation_id as the default
```

Comment:
@jameinel does this make sense?

Reply:
It doesn't make sense to grab it from a relation-id if relation-id was not supplied.

The logic would be (I think):
    look for the most specific match (binding, relation_id)
    then the next (binding, None)
    then the next <default>


but I wouldn't have (binding, None) return a match for (binding, relation_id).

Because the relation_id part would be an override for a particular case.


So I think we can actually simplify this a fair bit:

```
data = self._networks.get((binding_name, relation_id))
if data is not None:
  return data
if relation_id is not None:
  data = self._networks.get((binding_name, None))
  if data is not None:
    return data
# no custom data per relation id or binding, return the default binding
data = self._networks.get(("", None))
```

Reply:
That's really helpful, thanks John. I've updated it.